src/main/java/rex/Rex.java

```diff
@@ -0,0 +1,151 @@
+package rex;
+
+import rex.command.Command;
+import rex.exception.InvalidInputException;
+import rex.task.Deadline;
+import rex.task.Event;
+import rex.task.Task;
+import rex.task.TaskList;
+import rex.task.ToDo;
+import rex.util.Parser;
+import rex.util.Storage;
+import rex.util.Ui;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The {@code Rex} class serves as the main entry point for the Rex application,
+ * a task management system that allows users to manage their tasks via various commands.
+ * It integrates storage, task management, and user interaction functionalities.
+ */
+public class Rex {
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;

```

Comment:
I like your naming. Should taskList be in plural?

Reply:
Thank you for the suggestion, will keep that in mind when making changes to my code :)